File: common/src/main/java/dev/mayaqq/estrogen/registry/blocks/fluids/EstrogenLiquidBlock.java
package dev.mayaqq.estrogen.registry.blocks.fluids;

import dev.mayaqq.estrogen.registry.EstrogenEffects;
import earth.terrarium.botarium.common.registry.fluid.BotariumLiquidBlock;
import earth.terrarium.botarium.common.registry.fluid.FluidData;
import net.minecraft.core.BlockPos;
import net.minecraft.world.effect.MobEffectInstance;
import net.minecraft.world.entity.Entity;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.level.Level;
import net.minecraft.world.level.block.state.BlockState;

@SuppressWarnings({"deprecation", "NullableProblems"})
public class EstrogenLiquidBlock extends BotariumLiquidBlock {
    public EstrogenLiquidBlock(FluidData data, Properties properties) {
        super(data, properties);
    }

    @Override
    public void entityInside(BlockState state, Level level, BlockPos pos, Entity entity) {
        if (entity instanceof Player player) {
            player.addEffect(new MobEffectInstance(EstrogenEffects.ESTROGEN_EFFECT.get(), 20, 0));
        }
    }
}

File: src/main/kotlin/dev/mayaqq/estrogen/content/blocks/fluid/EstrogenLiquidBlock.kt
package dev.mayaqq.estrogen.content.blocks.fluid

import com.teamresourceful.resourcefullib.common.fluid.data.FluidData
import dev.mayaqq.estrogen.content.EstrogenEffects
import invoke.kitty.kritter.registry.api.entry.holder
import net.minecraft.core.BlockPos
import net.minecraft.world.effect.MobEffectInstance
import net.minecraft.world.entity.Entity
import net.minecraft.world.entity.LivingEntity
import net.minecraft.world.level.Level
import net.minecraft.world.level.block.state.BlockState

class EstrogenLiquidBlock(data: FluidData, properties: Properties, interactions: Array<FluidInteraction>) : BaseEstrogenLiquidBlock(data, properties, interactions) {
    override fun entityInside(state: BlockState, level: Level, pos: BlockPos, entity: Entity) {
        if (entity is LivingEntity) entity.addEffect(MobEffectInstance(EstrogenEffects.Estrogen.holder, 20, 0))
        super.entityInside(state, level, pos, entity)
    }
}